agent/pkg/providers/stats_provider.go

```diff
@@ -113,6 +119,97 @@ func GetAccumulativeStats() []*AccumulativeStatsProtocol {
 	return finalResult
 }
 
+func GetAccumulativeStatsTiming(intervalSeconds int, numberOfBars int) []*AccumulativeStatsProtocolTime {
+	bucketStatsCopy := BucketStats{}
+	if err := copier.Copy(&bucketStatsCopy, bucketsStats); err != nil {
```

Comment:
should this section be locked? to make sure no changed to the bucketsStats during the copy

Reply:
I have changed those lines to be in function with https://github.com/up9inc/mizu/pull/1157/commits/5909395b053adc932ee67e2c4547b98bc77962a8 so both places that use the object will get it from this function